nexus/db-model/src/vpc_route.rs

```diff
@@ -127,6 +127,27 @@ impl RouterRoute {
             destination: RouteDestination::new(params.destination),
         }
     }
+
+    pub fn for_subnet(
+        route_id: Uuid,
+        system_router_id: Uuid,
+        subnet: Name,
+    ) -> Result<Self, ()> {
+        let name = format!("sn-{}", subnet).parse().map_err(|_| ())?;
```

Comment:
Commenting for posterity, Kyle and I discussed this offline:

There seems to be a few ways conflicts / collisions can happen when attempting to apply nexus state to the sleds. We've seen similar issues with reconciling NAT state in Dendrite. With Dendrite, "whatever Nexus says, wins" so we have some logic that detects and removes entries that conflict with the Nexus request. While this can lead to undesired outcomes, it will at least result in consistent outcomes that we can understand and mitigate with more robust logic.

However it's likely that this situation may not lend itself to as simple a solution, and the idea of using mapping the user / operator provided subnet names to internal UUIDs, then using those internal UUIDs as the name when configuring the OPTE router instead of `sn-{}` could be a viable option.

Reply:
Fixed in 70263a2f3 by using `"{subnet}"` in place of `"sn-{subnet}"`. Here, we have the luxury of knowing that the only names we're going to collide with are "default-v4" and "default-v6" (internet gateway routes) -- users can alter their target, but not their name or destination in #5823. If someone happens to name their subnet identically, then the route will be named `"conflict-{subnet}-{uuid}"` which will be constrained below 63 characters.

Reply:
While mupdating to #5823 on london, it's become apparent there's some issue in the migration coming from this change. Will fix, and then we should be good to go.

Reply:
Caught it in 62de75e.